src/sources/docker_logs.rs

```diff
@@ -546,23 +522,20 @@ impl DockerLogsSource {
     /// True if container with the given id and image must be excluded from logging,
     /// because it's a vector instance, probably this one.
     fn exclude_vector<'a>(&self, id: &str, image: impl Into<Option<&'a str>>) -> bool {
-        if self.exclude_self {
-            let hostname_hint = self
-                .hostname
-                .as_ref()
-                .map(|maybe_short_id| id.starts_with(maybe_short_id))
-                .unwrap_or(false);
-            let image_hint = image
-                .into()
-                .map(|image| image.starts_with(VECTOR_IMAGE_NAME))
-                .unwrap_or(false);
-            if hostname_hint || image_hint {
-                // This container is probably itself.
-                info!(message = "Detected self container.", id = %id);
-                return false;
-            }
-        }
-        true
+        // Find out it's own container id, if it's inside a docker container.
+        // Since docker doesn't readily provide such information,
+        // various approaches need to be made. As such the solution is not
+        // exact, but probable.
+        let match_hostname = self
+            .hostname
+            .as_ref()
+            .map(|maybe_short_id| id.starts_with(maybe_short_id))
+            .unwrap_or(false);
+        let match_image = image
+            .into()
+            .map(|image| image.starts_with(VECTOR_IMAGE_NAME))
+            .unwrap_or(false);
+        match_hostname || match_image
```

Comment:
If I'm reading this right, this would exclude all containers on the local host. Did you mean `match_hostname && match_image`?

Reply:
Hi, `match_hostname` is true only if hostname is a short container ID (current vector process is inside a container, and the hostname of this cointainer matches the passed-in ID).